src/ensembl/doc/architecture/decisions/0016-server-side-rendering.md

```diff
@@ -0,0 +1,149 @@
+# 16. Server-Side Rendering
+
+## Status
+Under review.
+
+## Definitions
+- **Server-Side Rendering** means generating an HTML string representation of a web page on the server. The HTML string is sent over to the web browser, where it is parsed into a DOM tree and displayed on user's screen. The HTML string can be generated on the server dynamically during every request, or once during the build time. The latter is known as pre-rendering, or static site generation.
+- **Client-Side Rendering** means building the DOM tree of a web page directly in the browser, by executing a script that the browser receives from the server.
+
+## Context
+As of the time of writing this document, the new Ensembl site is rendered entirely client-side, which means that no matter which page is requested, the response is always a minimal html payload with a link to the bootstrap javascript file. That file, in its turn, will trigger the download of the rest of the required javascript files, which will then fetch the necessary data, and finally show the contents of the page on the screen.
+
+## Motivation
+### Shortcomings of client-side rendering
+#### Loss of control over HTTP status codes
+With client-side rendering, the browser always gets the same html payload, in a response with an HTTP status code `200: OK`. Thus, there is no way to signal, from the server, the intent to redirect the browser to a different page (300-range codes) or to message that the requested page does not exist (400-range errors).
+
+_**Note:** Use of improper http status codes can negatively impact search engine ranking by creating what Google calls "soft 404 errors" (404-error pages with status code 200). Google’s advice for client-side-rendered sites in this case is either to do a client-side redirect to a different url containing an actual 404-error page, or to add a `<meta name="robots" content="noindex">` tag to the page in order to prevent search engine's web crawler from indexing it._
+
+#### Negative impact on search engine discoverability and ranking
```

Comment:
Search engine visibility is considered strategically important

Reply:
The reason I put search visibility second is because:

- Google is capable of indexing purely client-side-rendered sites
- Even if we server-side-render, the only thing we should be able to really guarantee is that the html in the response will contain the appropriate meta tags in the `head` of the html document (title, description, maybe ld-json if there exists a schema on schema.org fitting for our use case). For pages that are mostly text, e.g. help&docs, we can easily make a case that they should be rendered server-side; but entity viewer, which is strategically the most important, is tricky.

But I'll move this section up, sure.

Reply:
Moved this section above the previous.